Documentation/internals/cilium_operator.rst

```diff
@@ -220,3 +220,22 @@ constructs to derivative CNP/CCNP objects without these fields.
 
 For more information, see how Cilium network policies incorporate the
 use of ``toGroups`` to :ref:`lock down external access using AWS security groups<aws_metadata_with_policy>`.
+
+Ingress and Gateway API Support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When Ingress or Gateway API support is enabled, the Cilium Operator performs the
+task of parsing Ingress or Gateway API objects and converting them into
+``CiliumEnvoyConfig`` objects used for configuring the per-node Envoy proxy.
+
+Additionally, Secrets used by Ingress or Gateway API objects will be synced to
+a Cilium-managed namespace that the Cilium Agent is then granted access to. This
+reduces the permissions required of the Cilium Agent.
+
+Mutual Authentication Support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When Cilium's Mutual Authentication Support is enabled, the Cilium Operator is
+responsible for ensuring that each Cilium Identity has an associated identity
+in the certificate management system (Currently only SPIRE is supported). It will
+create and delete SPIRE identity registrations as required.
```

Comment:
smal nit could we add that he operator itself has no access to the key material in the identities?